dataland-frontend/tests/component/testUtils/keycloak.ts

```diff
@@ -0,0 +1,30 @@
+import Keycloak from "keycloak-js";
+
+export interface KeycloakMockConfiguration {
+  userId?: string;
+  roles?: Array<string>;
+}
+
+/**
+ * A function returning a minmal keycloak mock sufficient for tricking the dataland frontend.
+ *
+ * @param config some configuration options that specify the user you want to imitate
+ * @returns a mocked keycloak object sufficient to trick the dataland frontend
+ */
+export function minimalKeycloakMock(config: KeycloakMockConfiguration): Keycloak {
+  const mock = {
+    token: "mocked-token",
+    authenticated: true,
+    idTokenParsed: {
+      sub: config.userId || "mock-user-id",
+    },
+    realmAccess: {
+      roles: config.roles || ["ROLE_USER"],
+    },
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    updateToken(minValidity: number): void {
```

Comment:
Why do we need this unused function?

Reply:
Its not really unused, eslint juts doesn't know its being used. The updateToken method is used in the ApiClients. As such the keycloak-mock must also provide it.

Reply:
And since the ApiClients are autogenerated code we do not want to include it into the code base for eslint checks. Hence, eslint will never know about it but we still need it. So the eslint-disable in this case makes sense.